website/docs/how-it-works.md

```diff
@@ -0,0 +1,86 @@
+---
+sidebar_position: 4
+---
+
+# How it Works
```

Comment:
We should have a larger section on the 3 different places we are involved (excited for this PR to go out so I can start adding too):
1. Compile (typescript to AST)
2. Synth (permissions, references, generation, AST -> language)
3. Runtime (client injection, serialization, opinionated language features, managed integrations)